### PR TITLE
Build a single library targeting a specific supported configuration

### PR DESCRIPTION
**Pull request by karajas**
Resolves https://github.com/dotnet/core-eng/issues/41

1. msbuild /p:BuildConfiguration=netstandard
2. Select netstandard-Windows_NT in configuration dropdown from VS and build.

@weshaggard @chcosta @ericstj

### CODE REVIEW

**Comment by karajas** [comment]
I believe the sln files may need updating as well.

**Comment by mellinoe** [comment]
I chatted with @karajas yesterday evening and we explored an alternative option, which looks like this:

* Remove all these "dummy blocks" from all project files
* Add every possible "dummy block" to dir.props or some other central place
* Adapt the tool/targets used for this PR to auto-generate sln files instead
* Check in an initial set of re-generated sln files

File: src/Microsoft.CSharp/src/Microsoft.CSharp.csproj [hunk 1]
```diff
@@ -9,9 +9,11 @@
     <ExcludeLocalizationImport Condition="'$(TargetsWindows)' != 'true'">true</ExcludeLocalizationImport>
   </PropertyGroup>
   <!-- Default configurations to help VS understand the configurations -->
-  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
+  <PropertyGroup Condition = "'$(Configuration)|$(Platform)' == 'netstandard_Debug|AnyCPU'" >
+    <BuildConfiguration>netstandard</BuildConfiguration>
```
Comment (weshaggard) [inline src/Microsoft.CSharp/src/Microsoft.CSharp.csproj]: You shouldn't need to set BuildConfiguration in these blocks. When building an individual project the Configuration and Platform should be enough. We might need to do some initialization of BuildConfiguration in a common location so the bin-placing works as expected.

**Comment by karajas** [comment]
Just to update, I've regened all the sln files and have them building and running src and test.

Talking with @ericstj , we probably want to impose a build order within the sln files to ensure that we always run the lastest bits. For documentation purposes, the order should be the following.
- source depends on refs 
- test depends on refs
- some tests depend on source (in the case of InternalsVisibleTo, and those that require to run with System.Private.CoreLib)
- Running of tests depends on tests and source

For now, the sln file has only the src and test since one of the requirements is that you rebuild all refs whenever you have a change.

File: src/Microsoft.CSharp/src/Microsoft.CSharp.csproj (hunk 1 above)
Comment (karajas) [inline src/Microsoft.CSharp/src/Microsoft.CSharp.csproj]: Removed in latest set.

**Comment by mellinoe** [comment]
How did you regenerate the sln files? Is it something like a tool we can include in buildtools? Semi-relatedly, are we going to be able to remove the configuration blocks from individual projects, or should we leave that as a further improvement step?

**Comment by karajas** [comment]
I have a tool for generating an updated sln file given an existing one. I plan on including like the opt in target mentioned so that the files can stay in sync. I'll try cleaning up the empty propety group's but afaict it's still needed to build.

File: build.proj [hunk 2]
```diff
@@ -76,6 +76,22 @@
     <MSBuild Projects="@(ExternalProject)"
              ContinueOnError="ErrorAndContinue" />
   </Target>
+  
+ <PropertyGroup>
```
Comment (weshaggard) [inline build.proj]: I assume you are planning to remove this before committing.

**Comment by weshaggard** [comment]
I agree with the build ordering in the solutions. We should just do the solution dependency ordering and not project references though. I suspect that is what you are referring to but I wanted to make sure.

**Comment by weshaggard** [comment]
go ahead and close this and make the PR target master

**Comment by karajas** [comment]
Will do

**Pull request closed without merging**